File: Problems/1636. Sort Array by Increasing Frequency.java
class Solution {
    public int[] frequencySort(int[] nums) {
        
        int freq[]=new int[201];
        
        for(int i:nums){
            
            freq[i+100]++;
            
        }
        
        int i=0;
        
        while(i!=nums.length){
            
            int smallest=Integer.MAX_VALUE,index=0;
            
            for(int j=200;j>=0;j--){
                if(freq[j]<smallest && freq[j]!=0){
                    smallest=freq[j];
                    index=j;
                }
            }
            
            while(smallest!=0){
                nums[i++]=index-100;
                --smallest;
            }
            
            freq[index]=0;
            
        }
        
        return nums;
        
    }
}

File: Problems/1636. Sort Array by Increasing Frequency.cpp
class Solution {
public:
    vector<int> frequencySort(vector<int>& nums) {
        
        int freq[201];

        for(int i: nums){
            freq[i+100]++;
        }

        vector<int> ans;

        while(ans.size() != nums.size()){

            int minFreq=101, smallestValue=0;

            for(int i=200; i>=0; i--){

                if(freq[i]!=0 && freq[i]<minFreq){
                    minFreq=freq[i];
                    smallestValue=i-100;
                }

                // cout<<"min freq="<<minFreq;
            }

            for(int i=0;i<minFreq;i++){
                ans.push_back(smallestValue);
                // cout << "pushing" << smallestValue;
            }

            freq[smallestValue+100]=0;

        }

        return ans;
    }
};